source/configuration_manual/mail_location/Maildir/index.rst

```diff
@@ -0,0 +1,148 @@
+.. _maildir_settings:
+
+=====================
+Maildir Configuration
+=====================
+
+See :ref:`Maildir <maildir_mbox_format>` for a technical description of how
+Dovecot has implemented Maildir support.
+
+Mail Location
+^^^^^^^^^^^^^
+
+Maildir exists almost always in ``~/Maildir`` directory. The mail location is
+specified with:
+
+.. code-block:: none
+
+  mail_location = maildir:~/Maildir
+
+Directory Layout
+^^^^^^^^^^^^^^^^
+
+By default, Dovecot uses Maildir++ directory layout. This means that all
+mailboxes are stored in a single directory and prefixed with a dot. For
+example:
+
+* Maildir/.folder/
+* Maildir/.folder.subfolder/
+
+If you want Maildirs to use hierarchical directories, such as:
+
+* Maildir/folder/
+* Maildir/folder/subfolder/
+
+you'll need to enable fs layout:
+
+.. code-block:: none
+
+  mail_location = maildir:~/Maildir:LAYOUT=fs
+
+.. _maildir_settings_control_files:
+
+Control Files
+^^^^^^^^^^^^^
+
+Dovecot stores some Maildir metadata into two control files:
+
+* ``dovecot-uidlist`` file contains IMAP UID <-> Maildir filename mapping
+* ``dovecot-keywords`` file contains Maildir filename flag (a..z = 0..25) <->
+  keyword name mapping
+
+Both of these files are described fully in :ref:maildir_mbox_formats. The
+important thing to remember about them is that they shouldn't be treated the
+same way as index files. Index files can be deleted and rebuilt without any
+side effects, but if you delete control files you'll cause messages to get
+new UIDs and possibly lose keyword names.
+
+If the messages get new UIDs, the IMAP clients will invalidate their local
+cache and download the messages all over again. If you do this for all the
+users, you could cause huge disk I/O bursts to your server.
+
+Dovecot cannot currently handle not being able to write the control files, so
+it will cause problems with `filesystem quota`_. To avoid problems with this,
+you should place control files into a partition where quota isn't checked. You
+can specify this by adding ``:CONTROL=<path>`` to ``mail_location``:
+
+.. code-block:: none
+
+  mail_location = maildir:~/Maildir:CONTROL=/var/no-quota/%u
+
+.. _`filesystem quota`: https://wiki.dovecot.org/Quota/FS
+
+Index Files
+^^^^^^^^^^^
+
+See :ref:`Mail Location Index Files` for a full explanation of how to change
+the index path. Example:
+
+.. code-block:: none
+
+  mail_location = maildir:~/Maildir:INDEX=/var/indexes/%u
+
+Optimizations
+^^^^^^^^^^^^^
+
+* ``maildir_very_dirty_syncs = yes``: Assume that only Dovecot accesses
```

Comment:
```
:ref:`maildir_very_dirty_syncs = yes <setting-maildir_very_dirty_syncs>`
```

Reply:
In order to be consistent with other changes, the opposite was done: maildir settings were moved from "core" page to maildir specific page.  Additionally, the information in the optimization section should instead live in the setting description, so make the optimization setting nothing more than the preferred settings and links to the settings page.

We discussed previously about differing views of where to put the internal settings links.  I feel 1) settings should live on the best page they can (core settings page is intended to list all settings, and it is the default home if there is not a better place for the settings to live) and 2) if settings live on a non-core page, any link should go directly to the setting itself, not to a placeholder on the core settings page where a user would have to click ANOTHER link to actually see the setting.  We can discuss further whether this is the right choice, but for now this MR will use this paradigm.